gensim/models/phrases.py

```diff
@@ -227,12 +243,17 @@ def export_phrases(self, sentences, out_delimiter=b' ', as_tuples=False):
             then you can debug the threshold with generated tsv
         """
         for sentence in sentences:
-            s = [utils.any2utf8(w) for w in sentence]
             last_bigram = False
             vocab = self.vocab
             threshold = self.threshold
             delimiter = self.delimiter  # delimiter used for lookup
             min_count = self.min_count
+
+            if self.recode_to_utf8:
+                s = [utils.any2utf8(w) for w in sentence]
+            else:
+                s = [utils.any2utf8(w) for w in sentence] if self.is_input_bytes else list(sentence)
```

Comment:
Ah, you're talking about what to do if the user sends a different type of data to `learn_vocab` vs `__getitem__`?

That's an interesting question -- should we try some implicit conversions? Or prefer to fail explicitly?

I think I'd prefer to fail explicitly, because the combination of "user asked not to recode during training" and "we implicitly recode to utf8 anyway during inference" could lead to subtle errors. For example, what happens if the user sent latin2 bytestrings to train, but we silently recode to utf8 during inference?